File: generated-code/codecraft/swift/Sources/TransGenTest/Entity.swift
/// Game entity
public struct Entity {
    /// Entity's ID. Unique for each entity
    let id: Int32

    /// Entity's owner player ID, if owned by a player
    let playerId: Int32?

    /// Entity's type
    let entityType: EntityType

    /// Entity's position (corner with minimal coordinates)
    let position: Vec2Int

    /// Current health
    let health: Int32

    /// If entity is active, it can perform actions
    let active: Bool

    /// Read Entity from input stream
    static func readFrom<S: InputStream>(_ stream: S) -> Entity {
        var id: Int32
        id = stream.readInt32()
        var playerId: Int32?
        if stream.readBool() {
            playerId = stream.readInt32()
        } else {
            playerId = nil
        }
        var entityType: EntityType
        entityType = EntityType.readFrom(stream)
        var position: Vec2Int
        position = Vec2Int.readFrom(stream)
        var health: Int32
        health = stream.readInt32()
        var active: Bool
        active = stream.readBool()
        return Entity(id: id, playerId: playerId, entityType: entityType, position: position, health: health, active: active)
    }

    /// Write Entity to output stream
    func writeTo<S: OutputStream>(_ stream: S) {
        stream.writeInt32(id)
        if playerId == nil {
            stream.writeBool(false)
        } else {
            stream.writeBool(true)
            let playerIdValue = playerId!
            stream.writeInt32(playerIdValue)
        }
        entityType.writeTo(stream)
        position.writeTo(stream)
        stream.writeInt32(health)
        stream.writeBool(active)
    }
}

File: generated-code/scala/codecraft/FileReadWrite/src/main/scala/trans_gen_test/model/Entity.scala
package trans_gen_test.model

import trans_gen_test.util.StreamUtil

/**
 * Game entity
 *
 * @param id Entity's ID. Unique for each entity
 * @param playerId Entity's owner player ID, if owned by a player
 * @param entityType Entity's type
 * @param position Entity's position (corner with minimal coordinates)
 * @param health Current health
 * @param active If entity is active, it can perform actions
 */
case class Entity(id: Int, playerId: Option[Int], entityType: trans_gen_test.model.EntityType, position: trans_gen_test.Vec2Int, health: Int, active: Boolean) {
    /**
     * Write Entity to output stream
     */
    def writeTo(stream: java.io.OutputStream) {
        StreamUtil.writeInt(stream, id)
        playerId match {
            case None => StreamUtil.writeBoolean(stream, false)
            case Some(value) => {
                StreamUtil.writeBoolean(stream, true)
                StreamUtil.writeInt(stream, value)
            }
        }
        entityType.writeTo(stream)
        position.writeTo(stream)
        StreamUtil.writeInt(stream, health)
        StreamUtil.writeBoolean(stream, active)
    }

    /**
     * Get string representation of Entity
     */
    override def toString(): String = {
        var stringBuilder = new StringBuilder("Entity { ")
        stringBuilder.append("id: ")
        stringBuilder.append(id)
        stringBuilder.append(", ")
        stringBuilder.append("playerId: ")
        stringBuilder.append(playerId)
        stringBuilder.append(", ")
        stringBuilder.append("entityType: ")
        stringBuilder.append(entityType)
        stringBuilder.append(", ")
        stringBuilder.append("position: ")
        stringBuilder.append(position)
        stringBuilder.append(", ")
        stringBuilder.append("health: ")
        stringBuilder.append(health)
        stringBuilder.append(", ")
        stringBuilder.append("active: ")
        stringBuilder.append(active)
        stringBuilder.append(" }")
        stringBuilder.toString()
    }
}

object Entity {
    /**
     * Read Entity from input stream
     */
    def readFrom(stream: java.io.InputStream): Entity = Entity(
        StreamUtil.readInt(stream),
        if (StreamUtil.readBoolean(stream)) Some(
            StreamUtil.readInt(stream)
        ) else None,
        trans_gen_test.model.EntityType.readFrom(stream),
        trans_gen_test.Vec2Int.readFrom(stream),
        StreamUtil.readInt(stream),
        StreamUtil.readBoolean(stream)
    )
}